src/main/java/edu/ucsb/mapache/controllers/RoleController.java

```diff
@@ -100,4 +108,41 @@
     String body = mapper.writeValueAsString(response);
     return ResponseEntity.ok().body(body);
   }
-}
+
+  // Added two functions below to modify DB to add API Token field
+  
+  @GetMapping("/apiKey")
+  public ResponseEntity<String> getCustomAPIToken(@RequestHeader("Authorization") String authorization)
+      throws JsonProcessingException {
+        AppUser user = authControllerAdvice.getUser(authorization);
+        String userToken = user.getApiToken();
+        Map<String, String> response = new HashMap<>();
+        response.put("token", userToken);
+        String body = mapper.writeValueAsString(response);
+        return ResponseEntity.ok().body(body);
+        
+  }
+
+  //Sets API Token
+  @PutMapping(value = "/addApiKey")
+  public ResponseEntity<String> setCustomApiToken(@RequestHeader("Authorization") String authorization, 
+  @RequestBody @Valid String token) 
+      throws JsonProcessingException {
+        AppUser user = authControllerAdvice.getUser(authorization);
+        SearchParameters sp = new SearchParameters();
+        sp.setQuery("empty");
+        sp.setPage(1);
+        
+        String i = googleSearchService.getJSON(sp, token);
+        if (i == "{\"error\": \"401: Unauthorized\"}") {
+          user.clearApiToken();
```

Comment:
If the new key that the user is trying to set isn't valid, we probably shouldn't be deleting the old key...

Reply:
Right now we're deleting the old key so that the user has a way to default back to the original key. Alternatively, we could have another button for the user to reset their token. Clearing the token = resetting back to default.

Reply:
I personally like the idea of having a "Clear Token" or "Reset to Default" button - that way, I don't have to worry about data being deleted when I don't want it to.

Reply:
I like that idea too.  But let's do that in a future issue.